tap_salesforce/client.py

```diff
@@ -315,15 +316,19 @@ def _login(self):
             )
         except requests.exceptions.HTTPError as req_ex:
             response_text = None
-            if req_ex.response:
+            if req_ex.response is not None:
                 response_text = req_ex.response.text
                 LOGGER.exception(response_text or str(req_ex))
-                if req_ex.response.status_code == 403:
-                    raise TapSalesforceOauthException(
-                        f"invalid oauth2 credentials: {req_ex.response.text}"
+
+                resp_json = req_ex.response.json()
+
+                if req_ex.response.status_code == 400 and resp_json.get("error") == "invalid_grant":
+                    raise TapSalesforceInvalidCredentialsException(
+                        f"invalid credentials: {response_text}"
```

Comment:
as we discussed, perhaps use the knowledge of how the object looks here, ie. if we have `error` and `message`, write the line as:
```python
f"invalid credentials: (error={resp_json['error']}, message={resp_json['message']})"
```

it would help the next person coming along to know how the object looks if an issue arises in this part of the code.

Reply:
thumbs up